packages/db/src/repositories/competition.ts

```diff
@@ -1579,6 +1581,110 @@ export class CompetitionRepository {
     }
   }
 
+  /**
+   * Calculate Sortino ratio metrics using database-level computations
+   * @param agentId Agent ID
+   * @param competitionId Competition ID
+   * @param mar Minimum Acceptable Return (default 0 for crypto)
+   * @returns Object with average return, downside deviation, and simple return
+   */
+  async calculateSortinoMetricsSQL(
```

Comment:
[nit] none of our methods end with `SQL` in the name (aside from `calculateMaxDrawdownSQL`), and we should type the returned object.